App/Controllers/LoginController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+
+namespace App\Controllers;
+
+use Core\View;
+use App\Models\User;
+use App\Models\Permission;
+
+class LoginController {
+
+    public function index()
+    {
+        $users = User::getDB()->readAll(User::getTableName(),User::class);
```

Comment:
Might be a good idea to create a Repository pattern logic here to fetch all users.

```
<?php

namespace App\Repositories\Interfaces;

use Illuminate\Database\Eloquent\Builder;

interface iGenericInterface
{
    public function getAll();

    public function find($id);

    public function findBy($field_name, $field_value);

    public function findAllBy($field_name, $field_value);

    public function getBasicQuery(): Builder;

    public function delete($id);
}
```

Create an AbstractDBRepository to have the common methods there

```
<?php

namespace App\Repositories\DB;

abstract class AbstractDBRepository{
 public static function getDB()
    {
        static $query = null;

        if ($query === null) {

            $pdo = Connection::make();

            $query = new QueryBuilder($pdo);
        }

        return $query;
    }
}

```

```
<?php

namespace App\Repositories\DB;

use App\Repositories\DB\Interfaces\iGenericInterface;
use App\Models\User;
use Core\QueryBuilder;

class UserRepository extends AbstractDBRepository implements iGenericInterface
{

    public function getAll($exclude_admin = false)
    {
        return self::getDB()->readAll(User::getTableName(),User::class);
    }

    public function find($id)
    {
        //apply here the relevant fetch
    }

    public function findBy($field_name, $field_value)
    {
        //apply here the relevant fetch
    }

    public function findAllBy($field_name, $field_value)
    {
        //apply here the relevant fetch
    }

    public function delete($id)
    {
        //apply here the relevant fetch
    }

}
```